### PR TITLE
Views section

### LINKED ISSUE #931: Person "Views"
"Views" would be a way to create custom tables ("views") for information tied to people, by adding columns using any type of data available in Zetkin, similarly (but opposite) to how the importer interface works. The view could then be used to display people using any existing method of filtering (e.g. Smart Search queries).

For example, adding columns "First name", "Last name", "E-mail", "Participated in campaign: My Campaign", "Has tag: My Tag" and "Response to survey question: 'What else do you want to do?'" would display a table with three text columns, two boolean columns (indicating campaign participation status and whether the person is tagged with the relevant tag) and then finally a text column showing the response to that survey question, if one exists.

The boolean columns could even be interactive, e.g. checkboxes.

A feature like this would solve many of the use cases where you would otherwise use spreadsheets.

### PR DESCRIPTION
**Pull request by richardolsson**
This PR adds a new Views section. Work in progress.

Closes #931 (see issue for design mock-up)

### CODE REVIEW

**Comment by richardolsson** [comment]
Remaining before ready to merge:

- [ ] Edit column title/description
- [ ] Delete column
- [ ] Delete view
- [ ] Limit length of text in cells (at least `survey_response`)

**Comment by richardolsson** [comment]
Postponing view delete to separate task #1115.

Hence ready for review!

**Comment by niklasva82** [comment]
This looks and works great!

Functionality-wise I would like to suggest a couple of things, not necessarily for now but in the future:
1. A way of starring all the people i a smart search. I sort of expected the star in the header to do this actually.
2. If a person is updated (e.g. tagged) in a Person pane, trigger a fetch of the corresponding row in order to update.

File: src/components/sections/Section.jsx
Comment (niklasva82) [inline src/components/sections/Section.jsx:67]: Replace the first pane? Where is this supposed to be used?

**Review by niklasva82 — CHANGES_REQUESTED** [review]
Overall it looks very good. I found two instances of misnamed classes which should be corrected.

**Comment by richardolsson** [comment]
> This looks and works great!
> 
> Functionality-wise I would like to suggest a couple of things, not necessarily for now but in the future:
> 
> 1. A way of starring all the people i a smart search. I sort of expected the star in the header to do this actually.
> 2. If a person is updated (e.g. tagged) in a Person pane, trigger a fetch of the corresponding row in order to update.

I agree about 2, and I have already implemented it (I believe I showed it at a weekly meeting recently?) locally. I was just waiting for this to be reviewed and merged before creating another pull request.

I think we have to give some more thought to 1. We don't want users to acidentally save/star thousands of rows, because it will completely overflow the view, obscuring whatever rows were saved before, and be very hard for the user to undo.

File: src/components/misc/personViews/cells/BooleanViewCell.jsx
Comment (richardolsson) [reply src/components/misc/personViews/cells/BooleanViewCell.jsx:5]: True! Will fix 😄

File: src/components/panes/AddViewColumnPane.jsx
Comment (richardolsson) [reply src/components/panes/AddViewColumnPane.jsx:40]: 🤦

File: src/components/sections/Section.jsx
Comment (richardolsson) [reply src/components/sections/Section.jsx:67]: This happens (via `gotoPane()`) when you select a view and the url changes from `views` to `views:1`.

See `PersonViewsPane.js:223`:

```jsx
<li key={ viewItem.data.id }
    className="PersonViewsPane-viewItem"
    onClick={ () => this.gotoPane('views', viewItem.data.id) }
    >
    <div className="PersonViewsPane-viewItemIcon"/>
    <span className="PersonViewsPane-viewItemTitle">{ viewItem.data.title }</span>
</li>
```

**Comment by richardolsson** [comment]
I've fixed the misnamed classes now. Ready for approval? 😄

**Review by niklasva82 — APPROVED** [review]

**Pull request merged**
